mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -505,7 +505,8 @@ object GeneralizedLinearRegression extends DefaultParamsReadable[GeneralizedLine
     override def initialize(y: Double, weight: Double): Double = {
       require(y >= 0.0, "The response variable of Poisson family " +
         s"should be non-negative, but got $y")
-      y
+      // Set lower bound for mean in the FIRST step in IWLS
```

Comment:
let's change this to `// force Poisson mean > 0 to avoid numerical instability in IRLS`